### PR TITLE
Add support for EP_AGORA

### PR DESCRIPTION
**Pull request by millerc-ep**

### CODE REVIEW

**Comment by millerc-ep** [comment]
@loverdeg-ep @trowbridgec

**Comment by 0xc0170** [comment]
How was this tested ? It would be helpful to provide logs as part of this PR

**Comment by 40Grit** [comment]
@0xc0170 This was done blind since we do not have an ac5 license.

Was hoping a build could be done via CI so we can use artifacts to test on our end.

File: test/info.py [hunk 1]
```diff
@@ -56,6 +56,7 @@
     ("k20dx_mimxrt1020_evk_qspi_if",                True,       0x8000,     "bin"       ),
     ("k20dx_mimxrt1050_evk_hyper_if",               True,       0x8000,     "bin"       ),
     ("k20dx_mimxrt1050_evk_qspi_if",                True,       0x8000,     "bin"       ),
+    ('k20dx_ep_agora_if',                           False,      0x0000,     "bin"       ),
```
Comment (trowbridgec) [inline test/info.py]: Do we need to match this to the other `k20dx` entries?

For example:
```
('k20dx_ep_agora_if',                           True,      0x8000,     "bin"       ),
```

**Comment by loverdeg-ep** [comment]
@maclobdell

File: test/info.py (hunk 1 above)
Comment (0xc0170) [reply test/info.py]: I would say yes

**Comment by 0xc0170** [comment]
> Was hoping a build could be done via CI so we can use artifacts to test on our end.

I can do. I'll start CI and share what I can with you

**Comment by 0xc0170** [comment]
2 errors in the build:

```
13:10:07 Compile [  1.3%]: ep_agora.c
13:10:07 [Error] ep_agora.c@28,21:  #154: expression must have struct or union type
13:10:07 [Error] ep_agora.c@29,16:  #57: this operator is not allowed in a constant expression
13:10:08 [ERROR] ".\source\board\ep_agora.c", line 28: Error:  #154: expression must have struct or union type
13:10:08       .infoVersion = 0x0002
13:10:08                      ^
13:10:08 ".\source\board\ep_agora.c", line 29: Error:  #57: this operator is not allowed in a constant expression
13:10:08       .board_id = "2600",
13:10:08                 ^
13:10:08 .\source\board\ep_agora.c: 0 warnings, 2 errors
```

File: test/info.py [hunk 2]
```diff
@@ -56,7 +56,7 @@
     ("k20dx_mimxrt1020_evk_qspi_if",                True,       0x8000,     "bin"       ),
     ("k20dx_mimxrt1050_evk_hyper_if",               True,       0x8000,     "bin"       ),
     ("k20dx_mimxrt1050_evk_qspi_if",                True,       0x8000,     "bin"       ),
-    ('k20dx_ep_agora_if',                           False,      0x0000,     "bin"       ),
+    ('k20dx_ep_agora_if',                           True,      	0x0000,     "bin"       ),
```
Comment (loverdeg-ep) [inline test/info.py]: Probably will want to fix the whitespace. Looks like a TAB too.

Comment (40Grit) [reply test/info.py]: @0xc0170 looking at this on my phone it doesn't appear like it did on my laptop

@millerc-ep is this ready for another build?

Comment (trowbridgec) [reply test/info.py]: @millerc-ep Spacing still seems a little off to me, might want to review just so it looks clean.

**Comment by 0xc0170** [comment]
I manually restarted the tests. Looks like CI does not report back status, have to check why not.

**Comment by 0xc0170** [comment]
The build completed, the EP_AGORA is not in the build artifacts , will investigate

**Comment by 0xc0170** [comment]
[k20dx_ep_agora_if.zip](https://github.com/ARMmbed/DAPLink/files/3591507/k20dx_ep_agora_if.zip)


Attached agora artifacts, please verify

**Comment by trowbridgec** [comment]
@0xc0170 Thank you!

The build looks good on our end...I was able to utilize drag and drop (MSD) functionality as well as debugging/flashing via pyocd with the `k20dx_ep_agora_if_crc_legacy_0x5000.bin` artifact on top of bootloader `0244_k20dx_bl_0x5000.bin` from the [here](https://os.mbed.com/blog/entry/DAPLink-bootloader-updates-with-external/).

**Comment by 0xc0170** [comment]
PR ready for integration?

**Comment by loverdeg-ep** [comment]
@0xc0170 I think it is ready. I would like to understand what the difference is between the "legacy" and not legacy build though.

**Comment by 0xc0170** [comment]
I expect test to fail, Mac machine has a problem with boards these days. I'll review if all the rest is OK

**Comment by 0xc0170** [comment]
All tests now green 💯

**Pull request merged**